### PR TITLE
[Baobzi] Add Baobzi_jll package (v0.9.6)

### PR DESCRIPTION
**Pull request by blackwer**
Adds build of https://github.com/flatironinstitute/baobzi shared lib.

### CODE REVIEW

File: B/Baobzi/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    Platform("i686", "linux"; libc = "musl"),
+    Platform("x86_64", "linux"; libc = "musl"),
+    Platform("aarch64", "linux"; libc = "musl"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("aarch64", "macos"; ),
+    Platform("x86_64", "freebsd"; ),
+    # Platform("i686", "windows"; ),
+    # Platform("x86_64", "windows"; )
+]
```
Comment (giordano) [inline B/Baobzi/build_tarballs.jl:45]:
```suggestion
platforms = supported_platforms(; exclude=Sys.iswindows)
```

File: B/Baobzi/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+script = raw"""
+cd $WORKSPACE/srcdir
+cmake -DCMAKE_INSTALL_PREFIX=$prefix \
+      -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DBAOBZI_BUILD_TESTS=OFF \
+      -DBAOBZI_BUILD_EXAMPLES=OFF \
+      -DBAOBZI_SET_ARCH=OFF \
+      baobzi
+make -j4
```
Comment (giordano) [inline B/Baobzi/build_tarballs.jl:23]:
```suggestion
make -j${nproc}
```

File: B/Baobzi/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+products = [
+    LibraryProduct("libbaobzi", :libbaobzi)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", preferred_gcc_version = v"12.1.0")
```
Comment (giordano) [inline B/Baobzi/build_tarballs.jl:58]: For maximum compatibility, we want to use the _oldest_ version of GCC that allows us to build the code.  Too recent versions are problematic: https://gcc.gnu.org/onlinedocs/libstdc++/manual/abi.html.

Comment (blackwer) [reply B/Baobzi/build_tarballs.jl]: I've had performance/c++ compatibility issues with older gcc versions in the past. From my limited testing, I think this is mostly a non-issue now. I've dropped the version as suggested.

**Comment by giordano** [comment]
Thanks for your contribution!

**Pull request merged**